pkg/api/index.go

```diff
@@ -125,6 +132,20 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 	return appLinks, nil
 }
 
+func (hs *HTTPServer) getDashboardUIDBySlug(c context.Context, orgID int64, slug string) (string, error) {
+	key := fmt.Sprintf("dashboard:%d:%s", orgID, slug)
```

Comment:
Rather than a new cache service, prefixing this key with `plugin_dashboard_by_slug` or similar should be enough I believe.